sftp-server.c

```diff
@@ -1641,6 +1645,39 @@ sftp_server_cleanup_exit(int i)
 	_exit(i);
 }
 
+void
+log_handler(LogLevel level, int forced, const char* msg, void* ctx)
+{
+	struct sshbuf* log_msg;
+	int* log_fd = (int*)ctx;
+	int r;
+	size_t len;
+
+	if (*log_fd == -1)
+		fatal_f("no log channel");
+
+	if ((log_msg = sshbuf_new()) == NULL)
+		fatal_f("sshbuf_new failed");
+
+	if ((r = sshbuf_put_u32(log_msg, 0)) != 0 || /* length; filled below */
```

Comment:
Reorder the way you send different properties.. 
Try to stick to mm_log_handler(). Add any new properties like pname, log_level, etc. in the end.